pkg/csi/service/wcp/controller.go

```diff
@@ -1339,23 +1339,23 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 							if len(nodeNames) == 1 && nodeNames[0] == req.NodeId {
 								log.Errorf("volume %q is still attached to node %q and podVM %q", req.VolumeId,
 									req.NodeId, podVM.Reference().String())
-								podvmpowerstate, err := podVM.PowerState(ctx)
-								if err != nil {
+								podvmpowerstate, powerstateErr := podVM.PowerState(ctx)
+								if powerstateErr != nil {
 									log.Errorf("failed to check the power state of pod vm: %q, error: %v",
 										podVM.Reference(), err)
-									return nil, csifault.CSIInternalFault, err
+									return nil, csifault.CSIInternalFault, powerstateErr
```

Comment:
I think returning an error like 'failed to get powerstate' may not be useful for end user in this case. We could return the error as the log on line 1340. 

`log.Errorf("volume %q is still attached to node %q and podVM %q", req.VolumeId, req.NodeId, podVM.Reference().String())`

Reply:
Addressed review comment

```
return nil, csifault.CSIInternalFault, fmt.Errorf("volume %q is still attached to "+
										"node %q and podVM %q Error while checking power state of Pod VM. Error: %v",
										req.VolumeId, req.NodeId, podVM.Reference().String(), powerstateErr)
```